From de73d6ae95902a6d9fb6140a2f64f76185b5da10 Mon Sep 17 00:00:00 2001
From: Dominic Hargreaves <dom@earth.li>
Date: Sun, 9 Jun 2013 11:38:52 +0100
Subject: Disable failing Net-Ping tests for GNU/Hurd

Not fully investigated, but best to have a working perl 5.18 on
GNU/Hurd for now.

Bug-Debian: http://bugs.debian.org/709385
Forwarded: no
Patch-Name: debian/hurd_net_ping_disable_test.diff
---
 dist/Net-Ping/t/450_service.t  |    2 +-
 dist/Net-Ping/t/510_ping_udp.t |    1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/dist/Net-Ping/t/450_service.t b/dist/Net-Ping/t/450_service.t
index 6c1d938..ca7b480 100644
--- a/dist/Net-Ping/t/450_service.t
+++ b/dist/Net-Ping/t/450_service.t
@@ -73,7 +73,7 @@ $p->{port_num} = $port2;
 
 {
     local $TODO;
-    $TODO = "Believed not to work on $^O" if $^O eq 'hpux';
+    $TODO = "Believed not to work on $^O" if $^O eq 'hpux' || $^O eq 'gnu';
     is($p->ping("127.0.0.1"), 1, 'second port is reachable');
 }
 
diff --git a/dist/Net-Ping/t/510_ping_udp.t b/dist/Net-Ping/t/510_ping_udp.t
index cb0ca1b..0a51c15 100644
--- a/dist/Net-Ping/t/510_ping_udp.t
+++ b/dist/Net-Ping/t/510_ping_udp.t
@@ -19,6 +19,7 @@ BEGIN {use_ok('Net::Ping')};
 SKIP: {
     skip "No udp echo port", 1 unless getservbyname('echo', 'udp');
     skip "udp ping blocked by Window's default settings", 1 if isWindowsVista();
+    skip "believed to be broken in Hurd (http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=709385)", 1 if $^O eq "gnu";
     my $p = new Net::Ping "udp";
     is($p->ping("127.0.0.1"), 1);
 }
